gg-pac-team3-back/src/controllers/userController.mjs

```diff
@@ -0,0 +1,178 @@
+const db = admin.firestore();
+const USERS = 'users'; 
+
+export const signupUser = async (req, res) => {
+    const { name, email, password, userType } = req.body;
+
+    if (!name || !email || !password) {
+        return res.status(400).json({ error: 'All fields are required' });
+    }
+
+    try {
+        const userRecord = await admin.auth().createUser({
+            name,
+            email,
+            password,
+            userType
+        });
+       
+        const defaultprofilePictureUrl = 'gs://learninghub-ggpacteam3.appspot.com/images/userProfileImage.jpg';
+        
+        const user = new UsersModel({
+            name,
+            email,
+            userType,
+            profilePictureUrl: defaultprofilePictureUrl
+        });
+
+        await db.collection(USERS).doc(userRecord.uid).set(user.toFirestore());
+
+        res.status(201).send({ message: 'User signed up successfully', user: userRecord });
+    } catch (error) {
+        console.error('Error signing up user:', error);
+        res.status(500).send({ error: error.message });
+    }
+};
+
+export const loginUser = async (req, res) => {
+    const { email, password } = req.body;
+
+    try {        
+        const apiKey = process.env.FIREBASE_API_KEY;        
+
+        const response = await axios.post(`https://identitytoolkit.googleapis.com/v1/accounts:signInWithPassword?key=${apiKey}`, {
+            email,
+            password,
+            returnSecureToken: true
+        });
+
+        const idToken = response.data.idToken;
+        const userRecord = await admin.auth().getUserByEmail(email);                
+
+        res.status(200).send({ message: 'User logged in successfully', token: idToken, user: userRecord });
+    } catch (error) {
+        console.error('Error logging in user:', error);
+        if (error.response) {
+            console.error('Error response data:', error.response.data);
+        }
+        res.status(400).send({ error: error.message });
+    }
+};
+
+export const getUser = async (req, res) => {
+    const { uid } = req.params;
+
+    try {
+        const userRecord = await admin.auth().getUser(uid);
+        res.status(200).send(userRecord);
+    } catch (error) {
+        console.error('Error fetching user:', error);
+        res.status(400).send({ error: error.message });
+    }
+};
+
+export const getAllUsers = async (req, res) => {
+    try {
+        const usersSnapshot = await db.collection(USERS).get();
+        if (usersSnapshot.empty) {
+            return res.status(404).send({ message: 'No users found' });
+        }
+        
+        const users = [];
+        usersSnapshot.forEach(doc => {
+            users.push({ id: doc.id, ...doc.data() });
+        });
+
+        res.status(200).send(users);
+    } catch (error) {
+        console.error('Error fetching users:', error);
+        res.status(500).send({ error: error.message });
+    }
+};
+
+export const getUserDashboard = async (req, res) => {
```

Comment:
What is this route for? It looks like it just gets a user object and returns it. How is this different than `GET /user/:userId`?

Reply:
"GET /user/:userId"  route  is for retrieving user information from firebase authentication and "GET /user/:userId/dashboard" route is to retrieves user profile data from a firestore collection.